templates/addons/linode-blockstorage-csi-driver-helm.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: addons.cluster.x-k8s.io/v1alpha1
+kind: HelmChartProxy
+metadata:
+  name: linode-blockstorage-csi-driver
+spec:
+  clusterSelector:
+    matchLabels:
+      csi-driver: linode
+  repoURL: https://linode.github.io/linode-blockstorage-csi-driver/
+  chartName: linode-blockstorage-csi-driver
+  namespace: kube-system
+  options:
+    waitForJobs: true
+    wait: true
+    timeout: 5m
+  valuesTemplate: |
```

Comment:
could we include this secret as part of the template?

Reply:
Having the Secret in the template would apply it to the management cluster, not the workload cluster. I opted to have the Secret creation be a manual step once the workload cluster is provisioned and reachable, mostly because of the API token requirement (region we could technically determine via `{{ .InfraCluster.spec.region }}` but this isn't anywhere in the spec and probably shouldn't be anyway). (https://github.com/linode/cluster-api-provider-linode/pull/115/files#r1490197122)

Reply:
I was able to get this working with the experimental/alpha [ClusterResourceSet](https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-resource-set).